symengine/symengine_assert.h

```diff
@@ -1,10 +1,21 @@
 #ifndef SYMENGINE_ASSERT_H
 #define SYMENGINE_ASSERT_H
 
-
+// SYMENGINE_ASSERT uses internal functions to perform as assert
+// so that there is no effect with NDEBUG
 #if !defined(SYMENGINE_ASSERT)
 #if defined(WITH_SYMENGINE_ASSERT)
-#define SYMENGINE_ASSERT(cond) assert(cond);
+#define stringize(s) #s
+#define XSTR(s) stringize(s)
+#define SYMENGINE_ASSERT(cond) \
+{ \
+if (0 == (cond)) { \
```

Comment:
Is there a reason why this cannot be written as `if (!(cond)) { \` ? Not a big deal, just asking.